browser/src/control/Control.QuickFindPanel.ts

```diff
@@ -33,13 +33,41 @@ class QuickFindPanel extends SidebarBase {
 		this.map.off('quickfind', this.onQuickFind, this);
 	}
 
+	addPlaceholderIfEmpty(quickFindData: any): any {
+		const hasEntries =
+			quickFindData.children &&
+			quickFindData.children.some(
+				(child: any) =>
+					child.type === 'treelistbox' &&
+					child.entries &&
+					child.entries.length > 0,
+			);
+
+		if (!hasEntries) {
+			const modifiedData = JSON.parse(JSON.stringify(quickFindData));
+
+			modifiedData.children.unshift({
+				id: 'quickfind-placeholder',
+				type: 'fixedtext',
+				text: _('Type in the search box to find anything in your document'),
+				visible: true,
+			});
+
+			return modifiedData;
+		}
+
+		return quickFindData;
+	}
+
 	onQuickFind(data: any) {
 		const quickFindData = data.data;
 
 		if (this.container) this.container.innerHTML = '';
 		else console.error('QuickFind: no container');
 
-		this.builder.build(this.container, [quickFindData], false);
+		const modifiedData = this.addPlaceholderIfEmpty(quickFindData);
```

Comment:
please do similar thing with onJSUpdate overriden from SidebarBase, then it should update always